quantized_mesh_encoder/extensions.py

```diff
@@ -0,0 +1,13 @@
+from dataclasses import dataclass
+
+from .constants import QuantizedMeshExtensions
+
+
+@dataclass
+class ExtensionBase:
+    extension_id: int
+
+
+@dataclass
+class VertexNormalsExtension(ExtensionBase):
+    extension_id: str = QuantizedMeshExtensions.VERTEX_NORMALS
```

Comment:
I think it might be cleanest to have an `encode` method on each extension. So for example:

```py
@dataclass
class VertexNormalsExtension(ExtensionBase):
	indices: np.ndarray
	positions: np.ndarray
    extension_id: str = QuantizedMeshExtensions.VERTEX_NORMALS

	def encode(self) -> bytes:
        normals = compute_vertex_normals(positions, indices)
        encoded = oct_encode(normals).tobytes('C')

        buf = b''
        buf += pack(
                EXTENSION_HEADER['extensionId'],
                QuantizedMeshExtensions.VERTEX_NORMALS)
        buf += pack(EXTENSION_HEADER['extensionLength'], len(encoded))
        buf += encoded
        return buf
```

This means that `positions` and `indices` will be given to this extension separately:
```py
ext = VertexNormalsExtension(indices, positions)
encode(..., extensions=[ext])
```

and within the global `encode` we can just use
```py
for ext in extensions:
    f.write(ext.encode())
```

Above is pseudocode, and I think you'd need to call `to_ecef` within this extension as well, but that's ok to duplicate because that function should be very fast. Though then you do need params to `to_ecef` as well... Maybe `encode` should take arguments like ellipsoid? I'm not sure.

Reply:
I think each extension dataclass should include all the data required to encode that extension, which would mean having ellipsoid as a parameter to the `VertexNormalsExtension`. Thoughts?

Reply:
Looks good to me! Yeah if we're explicitly setting `indices` and `positions` on the extension dataclass it makes sense to me for `ellipsoid` to also be a parameter. The only downside to this approach i guess is a slightly more verbose public API.

Reply:
Well, I think it cleans up the API for the top-level `encode`, because you don't need to pass detail extension parameters there; you can just pass extension dataclass objects. And then separately we document the API for each extension. And it would be very easy then to also implement the `metadata` and `watermask` extensions.

Reply:
Also, I can't remember: is it important for the normal creation to be done on cartesian positions instead of input positions? I assume yes?